spark-cassandra-connector/src/main/scala/org/apache/spark/sql/cassandra/CassandraConfCache.scala

```diff
@@ -0,0 +1,204 @@
+package org.apache.spark.sql.cassandra
+
+import com.datastax.spark.connector.cql.CassandraConnectorConf
+import com.datastax.spark.connector.rdd.ReadConf
+import com.datastax.spark.connector.writer.WriteConf
+import org.apache.commons.lang.StringUtils
+
+/**
+ * A local cache to store user defined Cassandra configuration settings. It includes
+ * per cluster level, per keyspace level and per table level settings. It also allows to
+ * update the settings. It's mainly used by Spark SQL, so user can define all kind of
+ * Cassandra configuration settings, and SQL query automatically get the settings to
+ * connect to, read from and write to Cassandra.
+ *
+ * Potentially it could be used by other components as well.
+ *
+ * If it's difficult to isolate the settings for your application, you still could pass
+ * explicitly those settings to the methods.
+ */
+
+class CassandraConfCache {
```

Comment:
Cache is very you store the info for later use instead of re-construct it. But you insist it's not a cache, then how about use name "ConfSetting"